Jackal/Jackal Shared/Playing Field/Shared/Level.swift

```diff
@@ -18,18 +18,58 @@ class Level {
     }
     
     // MARK: Public Properties
-    let graph: GKGridGraph<GKGridGraphNode>
+    let graph: BoardGraph<BoardGraphNode>
     
     
     // MARK: Lifecycle
     init(with size: Size) {
-        graph = GKGridGraph(fromGridStartingAt: vector_int2(x: 0, y: 0),
-                            width: size.width,
-                            height: size.height,
-                            diagonalsAllowed: true)
-        guard let nodes = graph.nodes else { return }
-        for node in nodes {
-            node.removeConnections(to: node.connectedNodes, bidirectional: true)
+        
+        graph = BoardGraph()
+        
+        for x in 0..<Int8(size.width) {
+            for y in 0..<Int8(size.height) {
+                
+                let position = BoardPosition(x, y)
+                let node = BoardGraphNode(boardPosition: position)
+                graph.add([node])
+            }
         }
+        
+        
+        for x in 0..<Int8(size.width) {
+            for y in 0..<Int8(size.height) {
+                
+                let position = BoardPosition(x, y)
+                guard let centre = graph.node(at: position) else { continue }
+                
+                let connectedPositions = [
```

Comment:
How do you see it in the future? Some kind of model will have the save matrix?

Reply:
yes